### DISCUSSION TITLE
New Sidebar component doesn't compile on vercel

### DISCUSSION THREAD

**Discussion by siconet** [shadcn-ui/ui#5500] [Q&A]
I just installed the new released awesome sidebar component and it works/compiles locally, but when deploying to vercel it has a build error:

Failed to compile.
./components/ui/sidebar.tsx:440:7
Type error: Type 'string | ((instance: HTMLDivElement | null) => void) | RefObject<HTMLDivElement> | null' is not assignable to type '((string | ((instance: HTMLDivElement | null) => void) | RefObject<HTMLDivElement>) & (((instance: HTMLElement | null) => void) | RefObject<...>)) | null | undefined'.
  Type 'string' is not assignable to type '((string | ((instance: HTMLDivElement | null) => void) | RefObject<HTMLDivElement>) & (((instance: HTMLElement | null) => void) | RefObject<...>)) | null | undefined'.
  438 |   return (
  439 |     <Comp
> 440 |       ref={ref}
      |       ^
  441 |       data-sidebar="group-label"
  442 |       className={cn(
  443 |         "duration-200 flex h-8 shrink-0 items-center rounded-md px-2 text-xs font-medium text-sidebar-foreground/70 outline-none ring-sidebar-ring transition-[margin,opa] ease-linear focus-visible:ring-2 [&>svg]:size-4 [&>svg]:shrink-0",
Error: Command "npm run build" exited with 1

![Bildschirmfoto 2024-10-22 um 06 44 27](https://github.com/user-attachments/assets/6a1fe03f-a8d9-412c-93d3-e1d378338c29)

**Comment by xDannyx** [comment]
Hi @siconet, have you found any workadound/fix for this build error?

**Reply by xDannyx** [reply]
I have found a solution here [https://github.com/shadcn-ui/ui/discussions/6080](url)

**Comment by siconet** [comment]
thank you for reching out. we dropped the protoype and started from scratch with material ui. but for the next prototype i am willing to use again next.js and shadcn-ui

Beste Grüße,

Dominik Rockenschaub

M: 0043/664 45 81 82 1
E: ***@***.***
Nachricht von xDannyx ***@***.***> am 17. Dez. 2024, 22:34 +0100:
